### PR TITLE
Refactors optionTapped func, adds guard to avoid crash.

### PR DESCRIPTION
**Pull request by valeriyvan**
Solves #3327.
As for me it doesn't solve origin of a problem.
For charts with the only dataset option to toggle dataset should't be shown at all.

### CODE REVIEW

File: ChartsDemo/Swift/Demos/LineChart1ViewController.swift [hunk 1]
```diff
@@ -142,35 +142,30 @@ class LineChart1ViewController: DemoBaseViewController {
     }
     
     override func optionTapped(_ option: Option) {
+        guard let sets = chartView.data?.dataSets as? [LineChartDataSet] else {
```
Comment (jjatie) [inline ChartsDemo/Swift/Demos/LineChart1ViewController.swift]: "{" on a new line please.

File: ChartsDemo/Swift/Demos/LineChart1ViewController.swift [hunk 2]
```diff
@@ -142,35 +142,30 @@ class LineChart1ViewController: DemoBaseViewController {
...
+            super.handleOption(option, forChartView: chartView)
+            return
+        }
+
         switch option {
         case .toggleFilled:
-            for set in chartView.data!.dataSets as! [LineChartDataSet] {
-                set.drawFilledEnabled = !set.drawFilledEnabled
-            }
+            sets.forEach{ $0.drawFilledEnabled = !$0.drawFilledEnabled }
```
Comment (jjatie) [inline ChartsDemo/Swift/Demos/LineChart1ViewController.swift]: a space between `forEach` and `{`

File: ChartsDemo/Swift/Demos/CombinedChartViewController.swift [hunk 3]
```diff
@@ -92,28 +92,27 @@ class CombinedChartViewController: DemoBaseViewController {
...
         switch option {
         case .toggleLineValues:
-            for set in chartView.data!.dataSets {
-                if let set = set as? LineChartDataSet {
-                    set.drawValuesEnabled = !set .drawValuesEnabled
-                    
-                }
-            }
+            data.dataSets.filter { $0 is LineChartDataSet }
+                         .forEach { $0.drawValuesEnabled = !$0.drawValuesEnabled }
```
Comment (jjatie) [inline ChartsDemo/Swift/Demos/CombinedChartViewController.swift]: This is fine here, but if you plan on contributing to the Charts framework avoid this. Here we have to loop through `dataSets` twice to enable toggle `drawValuesEnabled`, whereas something like :
```
for case let set as LineChartDataSet in data.dataSets {
    set.drawValuesEnabled = !set.drawValuesEnabled
}
```
only has to loop once.
I am curious what the performance characteristics of:
```
data.dataSets.lazy
    .flter { ... }
    .forEach { ... }
```
are like.

**Comment by valeriyvan** [comment]
done

File: ChartsDemo/Swift/Demos/CombinedChartViewController.swift (hunk 3 above)
Comment (valeriyvan) [reply ChartsDemo/Swift/Demos/CombinedChartViewController.swift]: I don't know an answer for your last question. 
Definitely, `sequence.filter{ $0.condition }.forEach{ $0.doSomething }` is slower than `sequence.forEach{ if $0.condition { $0.doSomething } }`. One is faster, another is readable. I guess, here, where we loop over data sets, minor performance loss could be sacrificed for readability.

**Comment by valeriyvan** [comment]
done done

**Comment by jjatie** [comment]
@valeriyvan take a look at my comment on #3327

**Comment by valeriyvan** [comment]
After putting so much effort into fixing bug the way you suggested, you change your mind(

**Comment by jjatie** [comment]
I know it sucks (it happens with my own PRs a lot). Sometimes you just need to see the outcome of a decision before you see it's wrong.

**Pull request closed without merging**